src/containers/BooksForm.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { createBook } from '../actions';
+import { bookCategories, generateID } from '../utils';
+import '../styles/BooksForm.css';
+
+
+class BooksForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      title: '',
+      category: 'Action',
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+
+  handleChange(event) {
+    const { name } = event.target;
+
+    this.setState({
+      [name]: event.target.value,
+    });
+  }
+
+  handleSubmit(event) {
+    const { title, category } = this.state;
+
+    // Validating the input form
+    if (title === '' || category === '') {
+      // eslint-disable-next-line
+      alert('Title can\'t be blank');
+      return;
+    }
+
+    event.preventDefault();
+
+    // eslint-disable-next-line
+    this.props.createBook({
+      id: generateID(),
+      title,
+      category,
+    });
+
+    this.setState({
+      title: '',
+      category: 'Action',
+    });
+  }
+
+  render() {
+    const bookOptions = bookCategories.map((category) => <option key={category} value={category}>{category}</option>);
+
+    return (
+      <div>
+        <h3>ADD NEW BOOK</h3>
+        <form onSubmit={this.handleSubmit}>
+          <input onChange={this.handleChange} className="input" type="text" name="title" value={this.state.title} placeholder="Title" />
```

Comment:
[OPTIONAL]
But actually I think this should be something that's included as crucial - we should deconstruct objects whenever there is a need to. For instance we should have deconstructed `this.state`:

```javascript
const { title, category } = this.state
```